evm/src/cpu/kernel/interpreter.rs

```diff
@@ -609,6 +609,32 @@ impl<'a> Interpreter<'a> {
         self.push(U256::from_big_endian(hash.as_bytes()));
     }
 
+    fn run_address(&mut self) {
+        self.push(
+            self.generation_state.memory.contexts[self.context].segments
+                [Segment::ContextMetadata as usize]
+                .get(ContextMetadata::Address as usize),
+        )
+    }
+
+    fn run_balance(&mut self) {
+        let offset = self.pop().as_usize();
```

Comment:
For this one I think we'd need to hash the address (to get the state trie key) and descend the trie based on that key. Maybe put a TODO for now?

(I don't think it's necessarily a requirement for the interpreter to support all opcodes - we use it for more lightweight unit tests, but we could decide to only exercise things like `BALANCE` in integration tests with the "real" zkEVM. So just calling it unsupported could be an option also.)

Reply:
Probably better to just close this PR for now I think, I believe there's additional tweaks to be done on some others, and it's indeed not a huge priority.

Reply:
I think the others appear fine to me, except the couple others that call `get_trie_data`

Reply:
I've left them as todo as you suggested for now, I may get back to them in a follow-up PR